File: java/src/model/ataque/GarraMecanica.java
package model.ataque;

import model.chimpos.Chinpokomon;

import java.util.Random;

public class GarraMecanica extends Ataque{

    public GarraMecanica() {
        super(2, "Garra Mecanica", 2);
    }

    public void atacar(Chinpokomon chipo, Integer valor) {
        Random random = new Random();
        int value = random.nextInt(10 + 1) + 1;
        if (value == 1) {
            chipo.recibirDanio(chipo.getVida() / 2);
        } else {
            chipo.recibirDanio(valor);
        }
    }
}

File: python/model/ataque/GarraMecanica.py
from model.ataque.Ataque import Ataque
from random import randint

class GarraMecanica(Ataque):
        
        def __init__(self, valor, nombre, valor_extra):
            Ataque.__init__(self, valor, nombre, valor_extra)
    
        def atacar(self, chipo, valor): 
            random = randint(1, 10)  
            if(random == 1):
                chipo.recibir_danio(chipo.vida / 2)
            else:
                chipo.recibir_danio(valor)